.dev/edge-native-base/src/oauth2_proxy/app.py

```diff
@@ -115,3 +124,33 @@ def oauth_callback():
         return response
 
     return app
+
+
+def ping_server():
+    """Send activity ping to the JupyterHub server"""
+
+    if not HAVE_JUPYTERHUB:
+        LOG.debug("Activity ping: skipped, no JupyterHub")
```

Comment:
It may be good to have throttling on pings. Perhaps the last_activity time can be stored in the session and a ping is sent only if it is still. If there is no throttling on pings then large applications that request dozens of files will send a ping on each individual file. 

Alternatively, this can be accomplished in nginx with a proxy_cache for the auth_status endpoint. This may be a little more difficult to implement but would be more performant.

https://serverfault.com/questions/897625/caching-nginx-auth-request